torch/testing/_internal/distributed/fake_pg.py

```diff
@@ -0,0 +1,17 @@
+import torch.distributed as dist
+
+# A fake process group (not related to FakeTensor) is a process group which
+# doesn't actually do any communication, it just hallucinates some
+# communication.  You can run a single rank with a fake process group
+# without needing multiple processes.
+
+class FakeProcessGroup(dist.ProcessGroup):
+    pass
+
+class FakeStore(dist.Store):
```

Comment:
we probably don't need a FakeStore and instead we can just use HashStore I suppose. But that would be in a separate PR.

Reply:
Hmmm I think FakePG barrier should be a no-op, and in terms of `init_processs_group` store based barrier it seems we already skip the barrier so it won't write anything to the HashStore. So either fake_store or hash_store could work I feel (I can give it a try and see if that's feasible or not)

Reply:
FSDP does a barrier which is why I ended up doing FakeStore. But yeah, try some stuff out, the goal is to be able to run FSDP end-to-end with the fake group with only one node.